Constraints support

chrisism opened the pull request:
Added support to set the value of variable when a certain constraint is met. Users can supply a regex and a certain value and only if that matches it will execute the set var. There is also an option to set an alternative value in case the constraint is not met.

jessehouwing commented on vsts-variable-set/vsts-variable-set.ps1:
```diff
@@ -8,20 +8,32 @@ param(
 $PSBoundParameters.Keys | %{ Write-Verbose "$_ = $($PSBoundParameters[$_])" }
 
-if ($IfVariable -ne "") 
+if ($applyConstraint.ToUpper() -eq "TRUE") 
```
Can you make this `-eq $true` that should work just fine.

jessehouwing commented on vsts-variable-set/vsts-variable-set.ps1:
```diff
@@ -8,20 +8,32 @@ param(
     {
-        Write-Verbose "Value not set. Could not match if constraint."
-        exit
+        if ($applyAlternativeValue -eq "FALSE")
```
Can you make this `-eq $false` that should work just fine.

jessehouwing commented on Build.bat:
```diff
@@ -0,0 +1 @@
+tfx extension create --manifest-globs vss-extension.json
```
Can you please amend the commit to remove the batch file from the commit instead of in a separate commit?

chrisism commented:
Yeah, I did a quick fork and push. More to let you know of some other possiblities with the stuff.
But I can ofcourse cleanup this branch a bit more that its ok with you to merge. Indeed there are far more possiblities, but I implemented only Regex because it was the quickest one to implement and it covers a lot of situations. More options would be ofcourse awesome.
I added it to Set var because I see it as an if/else type of Set. The reason I need it is because I have one CI build for a project that builds nuget packages. I also use GitVersion in this project. Depending on the branch that is being build I either want to make a prerelease package (using the full gitversion number with text) or I want a stable package (in case of master branche and then a full major.minor.revision.build etc). So now you know why.

Add more change requests so I know what to fix for the branch.
